### PR TITLE
VACMS 4110: Improve content release integration with AdvancedQueue Part Deux

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

See #4110.  My previous PR, #4543, is too broad and touches too many things, and some changes were obsoleted by the upgrade to Drush 10, so this is a more tightly-focused PR.  

## Testing done
Adding behavioral, unit, and kernel tests as needed.  Current tests aren't reliable or even correct, just a starting point.

## Screenshots
So in the normal state of things, the web build status is FALSE:
Screen Shot 2021-03-11 at 2 57 41 PM

and the web build queue is empty:
Screen Shot 2021-03-11 at 2 57 52 PM

With no jobs in the queue, running the queue will return (almost) immediately:
Screen Shot 2021-03-11 at 2 59 18 PM

The following command adds a build job to the queue...
Screen Shot 2021-03-11 at 3 00 05 PM

...  but does not run it.  It will remain in the "queued" state indefinitely.
Screen Shot 2021-03-11 at 3 00 45 PM

Running the queue will start the job in the foreground:
Screen Shot 2021-03-11 at 3 01 40 PM

In another console, we can see that the web build status has been set TRUE:

Screen Shot 2021-03-11 at 3 02 35 PM

and that the queue list shows the job as processing:
Screen Shot 2021-03-11 at 3 03 11 PM

If we kill the Job's current process...
Screen Shot 2021-03-11 at 3 04 22 PM

The foreground job will be terminated.

Screen Shot 2021-03-11 at 3 04 50 PM

The status will be correctly reset to FALSE:
Screen Shot 2021-03-11 at 3 05 53 PM

And the queue will show the failed job:
Screen Shot 2021-03-11 at 3 06 54 PM

If we issue another build request and run the queue again:
Screen Shot 2021-03-11 at 3 07 49 PM

We'll see the status re-set to TRUE and see the job processing in the queue list.
Screen Shot 2021-03-11 at 3 09 03 PM

We can submit another build job...
Screen Shot 2021-03-11 at 3 10 17 PM

And it will be listed in the queue: 
Screen Shot 2021-03-11 at 3 10 44 PM

But running the queue will return quickly, and, listing the queue again, we'll see the error message and that the job will be re-enqueued in a few seconds.
Screen Shot 2021-03-11 at 3 11 37 PM

At some point, the current job will complete successfully (🎉 ) and the next job will begin processing.
Screen Shot 2021-03-11 at 3 27 28 PM

Of course, the web build status will be TRUE:
Screen Shot 2021-03-11 at 3 29 18 PM

But at some point, the queue-run will complete:
Screen Shot 2021-03-11 at 3 41 10 PM

The second job will be shown as having completed successfully in the queue list:
Screen Shot 2021-03-11 at 3 41 00 PM

And the web build status will be set back to FALSE:
Screen Shot 2021-03-11 at 3 40 45 PM

## QA steps

As user _uid_ with _user_role_
1. Do this
1. Then that
1. Then validate Acceptance Criteria from issue #4110:
    - [ ] users can see the status of all builds on the Content Release page (Tugboat/local) regardless of how it was initiated **NOT CURRENTLY MET**: The `tests/scripts/build-web.sh` script will not executed via the queue.
    - [ ] advancedqueue runner checks for in-process web builds and requeues the task if any are encountered or if the build has errored
    - [ ] builds complete successfully regardless of how they were initiated
    - [ ] check if the state is removed when job errors
    - [ ] split composer commands into (1) runner (2) queuer

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [x] `Core Application Team`
- [ ] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [x] No announcement is needed for this code change. 
  - [x] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by ndouglas** [comment]
Weirdly, my image-crop tests are passing locally (whether I run the .feature individually or run all of them together).

**Comment by ndouglas** [comment]
D'oh.  Realized I missed the `tests/scripts/build-web.sh` call 🤔

**Comment by ndouglas** [comment]
I've been blasting away at this PR for so long, even my momma thinks that my mind is gone
maxresdefault

@indytechcook @acrollet I know you guys are like holy crap busy right now, but I'd appreciate any thoughts you have on this PR.  I don't think the existing automated tests are sufficient, but I feel like there's a substantial amount of effort left to implement them.  

I'm thinking that I should "mock" jobs with command payloads like `[ 'exit 0' ]`, `[ 'exit -43' ]`, `[ 'sleep 10; exit 0' ]`, and so forth, but wanted to see if you guys had any further suggestions, requests, etc.

This obviously ain't getting merged and deployed today, so my goal right now is to just get it done in a quasi-reasonable time frame and incurring as little tech debt as possible.  ❤️

**Review by acrollet — COMMENTED** [review]
the code looks great to me! I added a minor suggestion that's not at all essential.

File: composer.json [hunk 1]
```diff
@@ -499,46 +499,53 @@
...
         ],
         "va:web:build": [
-            "drush state:set va.web_build_stats TRUE -y",
             "rm -rf docroot/vendor/va-gov/web/build/vagovdev",
             "cd web && echo \"Downloading assets for vets-website ref $(git rev-parse --verify HEAD)\" ",
             "cd web && NODE_ENV=production yarn build:content --pull-drupal --no-drupal-proxy --buildtype=vagovdev --api=https://dev-api.va.gov --asset-source=$(git rev-parse --verify HEAD)",
             "echo 'Replace s3 address with local in generated files'",
-            "find ./docroot/vendor/va-gov/web/build/vagovdev/generated -type f -print0 | xargs -0 sed -i \"s/https:\\/\\/dev-va-gov-assets\\.s3-us-gov-west-1\\.amazonaws\\.com//g\" ",
-            "drush state:set va.web_build_stats FALSE -y"
+            "find ./docroot/vendor/va-gov/web/build/vagovdev/generated -type f -print0 | xargs -0 sed -i \"s/https:\\/\\/dev-va-gov-assets\\.s3-us-gov-west-1\\.amazonaws\\.com//g\" "
```
[the remark this answers is not in the document]
Comment (ndouglas) [reply composer.json:521]: Yeah, that's absolutely worth doing.  I'll check it interactively tomorrow just to make sure.  I never have a good grasp on what's PCRE and what's sed-friendly 😩

[the remark this answers is not in the document]
Comment (acrollet) [reply composer.json:521]: yeah me neither 😳

**Review by indytechcook — CHANGES_REQUESTED** [review]
I did a code revie with Nate on zoom and the only change is to remove the extra composer commands since they are just calling drush commands and to update documentation with the new drush commands.

Otherwise this is awesome.

File: composer.json [hunk 2]
```diff
@@ -522,28 +522,28 @@
             "cd web && echo \"Downloading assets for vets-website ref $(git rev-parse --verify HEAD)\" ",
             "cd web && NODE_ENV=production yarn build:content --pull-drupal --no-drupal-proxy --buildtype=vagovdev --api=https://dev-api.va.gov --asset-source=$(git rev-parse --verify HEAD)",
             "echo 'Replace s3 address with local in generated files'",
-            "find ./docroot/vendor/va-gov/web/build/vagovdev/generated -type f -print0 | xargs -0 sed -i \"s/https:\\/\\/dev-va-gov-assets\\.s3-us-gov-west-1\\.amazonaws\\.com//g\" "
+            "find /app/docroot/vendor/va-gov/web/build/vagovdev/generated -type f -exec sed -i \"s#https://dev-va-gov-assets\\.s3-us-gov-west-1\\.amazonaws\\.com##g\" {} \\+"
```
Comment (acrollet) [inline composer.json:525]: I don't think using `/app` will work on tugboat...

Comment (ndouglas) [reply composer.json:525]: good catch -- actually inserted that to test locally 😩

Comment (indytechcook) [reply composer.json]: This is also run on BRD fyi

**Review by acrollet — APPROVED** [review]
All feedback addressed, lgtm

**Comment by ndouglas** [comment]
https://www.youtube.com/watch?v=u_8nAvU0T5Y

**Review by indytechcook — APPROVED** [review]

**Pull request merged**